verified-reviews/utils/client.ts

```diff
@@ -0,0 +1,165 @@
+import { fetchAPI } from "../../utils/fetch.ts";
+import {
+	ConfigVerifiedReviews,
+	Ratings,
+	Reviews,
+	VerifiedReviewsFullReview,
+} from "./types.ts";
+import { Product } from "../../commerce/types.ts";
+
+export type ClientVerifiedReviews = ReturnType<typeof createClient>;
+
+export interface PaginationOptions {
+	count?: number;
+	offset?: number;
+	order?:
+		| "date_desc"
+		| "date_ASC"
+		| "rate_DESC"
+		| "rate_ASC"
+		| "helpfulrating_DESC";
+}
+
+const baseUrl = "https://awsapis3.netreviews.eu/product";
+
+export const createClient = (params: ConfigVerifiedReviews | undefined) => {
+	if (!params) return;
+
+	const { idWebsite } = params;
+
+	/** @description https://documenter.getpostman.com/view/2336519/SVzw6MK5#338f8f1b-4379-40a2-8893-080fe5234679 */
+	const rating = async ({ productId }: { productId: string }) => {
+		const payload = {
+			query: "average",
+			products: [productId],
+			idWebsite: idWebsite,
+			plateforme: "br",
+		};
+
+		try {
+			const data = await fetchAPI<Ratings>(`${baseUrl}`, {
+				method: "POST",
+				body: JSON.stringify(payload),
+			});
+
+			return Object.keys(data).length ? data : undefined;
+		} catch (error) {
+			console.warn("⚠ Error on call rating of Verified Review", error);
+			return undefined;
+		}
+	};
+
+	/** @description https://documenter.getpostman.com/view/2336519/SVzw6MK5#6d8ab05a-28b6-48b3-9e8f-6bbbc046619a */
+	const ratings = async ({ productsIds }: { productsIds: string[] }) => {
+		const payload = {
+			query: "average",
+			products: productsIds,
+			idWebsite: idWebsite,
+			plateforme: "br",
+		};
+		try {
+			const data = await fetchAPI<Ratings>(`${baseUrl}`, {
+				method: "POST",
+				body: JSON.stringify(payload),
+			});
+
+			return Object.keys(data).length ? data : undefined;
+		} catch {
+			console.warn(
+				"🔴⭐ Error on call ratings of Verified Review - probably unidentified product"
```

Comment:
Should we break in this case? I don't know if a warn is enough.


How about testing if it is production or locally?

tip: you can use `context.isDeploy` 

import { context } from "deco/live.ts"